framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
I would rather have this handled by the framework calling `shouldApply()` and then just _not_ calling `Predictor::apply()` if it's `false`...